### DISCUSSION TITLE
locale props with component isn't being render in 2.9.2 ?

### DISCUSSION THREAD

**Discussion by LatentDream** [gilbarbara/react-joyride#1107] [Q&A] [answered]
I have a tour, using:

```jsx
        <Joyride
            run={run}
            steps={steps}
            callback={callback}
            locale={{
                skip: <CustomSkipButton />
            }}
```

Where `CustomSkipButton` is a React component:
```jsx
function CustomSkipButton() {
    return (
        <StyledCustomSkipButton>
            Click here to end tutorial
        </StyledCustomSkipButton>
    );
}
```

Migrating to the latest version (2.9.2), the CustomSkipButton isn't being rendered anymore. Only If I use plain text , the skip button is being display, e.g.:
```jsx
        <Joyride
            run={run}
            steps={steps}
            callback={callback}
            locale={{
                skip: "Skip Button"
            }}
```

___

Did the API changed ? I still see ReactNode as the type input ? Or is this a bug ?

**Comment by gilbarbara** [comment]
Hey @LatentDream 

Nope, it's still working.
Check the [demo](https://codesandbox.io/p/devbox/github/gilbarbara/react-joyride-demo?file=%2Fsrc%2FBasic%2Findex.tsx)

**Comment by LatentDream** [comment]
Hey @gilbarbara, thanks for taking a look that quickly. Must be a something in our code base, I'll investigate further later today on why the switch to `2.9.1` to `2.9.2` does that

Thanks

**Reply by gilbarbara** [reply]
Yeah, I don't think that the changes in 2.9.2 would have an effect on locale. 
Good old remove node_modules and reinstall?

**Reply by LatentDream** [reply]
It's actually the text being stripped from the element, with a custom component:

[Example - Works on 2.9.1, but not on 2.9.2](https://codesandbox.io/p/devbox/fast-morning-yy456k?workspaceId=5b11d3e9-cc2d-4f4e-885f-0c859ea0bdc1)`

```
// Wrapping the <strong> in another element trigger the problem
function CustomSkipButton() {
  return <strong>This text won't be display in 2.9.2 (was display on 2.9.1)</strong>;
}

export default function BasicDemo(props: Props) {
  const { breakpoint } = props;
  const [{ run, steps }, setState] = useSetState<State>({
    run: false,
    steps: [
      {
        content: <h2>Let's begin our journey!</h2>,
        locale: { skip: <CustomSkipButton /> },
        placement: 'center',
        target: 'body',
      },
      {
        content: <h2>Let's begin our journey!</h2>,
        locale: { skip:  <strong>This text will be display in 2.9.2</strong> },
        placement: 'center',
        target: 'body',
      },
...
```

Related to [this changed](https://github.com/gilbarbara/react-joyride/commit/58ecae0dd400f7ba6d6241c13d35e133a4cea772#diff-7e5f12af97629920eda2d123a3861fc8602e39ed6363dfd2fe34e2da25c73e87R38-R41) ?

**Reply by gilbarbara** [reply]
Damn, I forgot to add tests for this scenario.
I'm working on it.

**Comment by gilbarbara** [comment, marked as answer]
Released as [2.9.3](https://github.com/gilbarbara/react-joyride/releases/tag/v2.9.3)
Thanks

**Reply by LatentDream** [reply]
Amazing! Thank you :)
